NEW: Improve Displaying Access Keys in Navigation

aljawaid opened the pull request:
## NEW: Improve Displaying Access Keys in Navigation
- Improved to show ~just the key~ the detected browser's full shortcut in arrow tooltip
- ~Improved to add keyboard shortcut instructions in help menu hover~
- Added `classfortooltip` for better styling

aljawaid committed:
Merge pull request #1 from Dolibarr/develop

Pull Latest

aljawaid committed:
FIX: Display All Types of Payments on Expense Report Card 

Fixes #9991 - Bug: 7.0.4: Expense Report Payment Type Not Displayed 
- without this fix, only cash type accounts were shown when payment was made

aljawaid committed:
Merge pull request #2 from aljawaid/aljawaid-fix_payment_type

FIX: Display All Types of Payments on Expense Report Card

aljawaid committed:
Add Access Keys in Help Link

aljawaid committed:
Add Access Keys in Navigation Tooltips

aljawaid committed:
Add Access Key Language Strings

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -6683,8 +6683,8 @@ function showrefnav($object,$paramid,$morehtml='',$shownav=1,$fieldid='rowid',$f
 			// accesskey is for Mac:               CTRL + key for all browsers
-			$previous_ref = $object->ref_previous?'<a accesskey="p" title="'.$langs->trans("KeyboardShortcut").' ALT+p|ALT+SHIFT+p|CTRL+p" href="'.$navurl.'?'.$paramid.'='.urlencode($object->ref_previous).$moreparam.'"><i class="fa fa-chevron-left"></i></a>':'<span class="inactive"><i class="fa fa-chevron-left opacitymedium"></i></span>';
-			$next_ref     = $object->ref_next?'<a accesskey="n" title="'.$langs->trans("KeyboardShortcut").' ALT+n|ALT+SHIFT+n|CTRL+n" href="'.$navurl.'?'.$paramid.'='.urlencode($object->ref_next).$moreparam.'"><i class="fa fa-chevron-right"></i></a>':'<span class="inactive"><i class="fa fa-chevron-right opacitymedium"></i></span>';
+			$previous_ref = $object->ref_previous?'<a accesskey="p" title="'.$langs->trans("AccessKey").' = p" href="'.$navurl.'?'.$paramid.'='.urlencode($object->ref_previous).$moreparam.'"><i class="fa fa-chevron-left"></i></a>':'<span class="inactive"><i class="fa fa-chevron-left opacitymedium"></i></span>';
```
Why no using $conf->browser->os to keep the ALT, or ALT + SHIFT depending on browser...

aljawaid replied:
I only changed the title, I wouldn't know where to put `$conf->browser->os`...

or do you mean... `title="'.$conf->browser->os.' = p"'` ?

eldy replied:
No i mean

$stringforfirstkey = 'CTL'
if ($conf->browser->name == 'chrome') $stringforfirstkey = 'ALT'
if ($conf->browser->name == 'firefox') $stringforfirstkey = 'ALT  SHIFT + KEY'

then
title="'.$stringforfirstkey.' = p"' ?

So user kown what is the complete shortcut.

aljawaid committed:
Undo Access Key Strings from previous commit

aljawaid committed:
Undo Add Access Keys in Help Link ... from previous commit

Not relevant anymore as the new code will detect browser and show the relevant string

aljawaid committed:
Amend Add Access Keys in Navigation Tooltips 

- based on suggested code by @eldy
- Added `classfortooltip` for better display

aljawaid commented:
@eldy done, thanks for your suggestion.

aljawaid committed:
Amend Add Access Keys in Navigation Tooltips

- Fix travis

eldy committed:
Update main.lang

eldy committed:
Update html.form.class.php
